samples/readingEvents.go

```diff
@@ -0,0 +1,246 @@
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion iterate-stream
+}
+
+func ReadFromStreamPosition(db *client.Client) {
+	// region read-from-position
+	ropts := options.ReadStreamEventsOptionsDefault().
+		Position(stream_position.Revision(10))
+
+	stream, err := db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 20)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+	// endregion read-from-position
+	// region iterate-stream
+	for {
+		event, err := stream.Recv()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion iterate-stream
+}
+
+func ReadFromStreamPositionCheck(db *client.Client) {
+	// region checking-for-stream-presence
+	ropts := options.ReadStreamEventsOptionsDefault().
+		Position(stream_position.Revision(10))
+
+	stream, err := db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 100)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+
+	for {
+		event, err := stream.Recv()
+
+		if err == errors.ErrStreamNotFound {
+			fmt.Print("Stream not found")
+		}
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion checking-for-stream-presence
+}
+
+func ReadSreamBackwards(db *client.Client) {
+	// region reading-backwards
+	ropts := options.ReadStreamEventsOptionsDefault().
+		Position(stream_position.End()).
+		Backwards()
+
+	stream, err := db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 10)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+
+	for {
+		event, err := stream.Recv()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion reading-backwards
+}
+
+func ReadFromAllStream(db *client.Client) {
+	// region read-from-all-stream
+	ropts := options.ReadAllEventsOptionsDefault().
+		Position(stream_position.Start()).
+		Forwards()
+
+	stream, err := db.ReadAllEvents(context.Background(), &ropts, 100)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+	// endregion read-from-all-stream
+	// region read-from-all-stream-iterate
+	for {
+		event, err := stream.Recv()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion read-from-all-stream-iterate
+}
+
+func IgnoreSystemEvents(db *client.Client) {
+	// region ignore-system-events
+	ropts := options.ReadAllEventsOptionsDefault()
+
+	stream, err := db.ReadAllEvents(context.Background(), &ropts, 100)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+
+	for {
+		event, err := stream.Recv()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+
+		if strings.HasPrefix(event.GetOriginalEvent().EventType, "$") {
```

Comment:
Oh, that's pretty magic and implicit. Is it part of the protocol/standard/whatever? Maybe we can have a getter in the event like `IsSystemEvent`?

Reply:
No, it's not but maybe we should :smile: